src/Keboola/DbExtractor/DbRetryProxy.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\DbExtractor;
+
+use Psr\Log\LoggerInterface;
+use Retry\RetryProxy;
+use Retry\Policy\RetryPolicyInterface;
+use Retry\Policy\SimpleRetryPolicy;
+use Retry\BackOff\BackOffPolicyInterface;
+use Retry\BackOff\ExponentialBackOffPolicy;
+
+class DbRetryProxy extends RetryProxy
+{
+    public const DEFAULT_MAX_TRIES = 5;
+
+    /* set initial backoff to 1 second */
+    public const DEFAULT_BACKOFF_INTERVAL = 1000;
+
+    public const DEFAULT_EXCEPTED_EXCEPTIONS = ['PDOException', 'ErrorException'];
+
+    public function __construct(
+        LoggerInterface $logger,
+        ?int $maxTries = null,
+        ?array $expectedExceptions = null,
+        ?int $backoffInterval = null,
+        ?RetryPolicyInterface $retryPolicy = null,
+        ?BackoffPolicyInterface $backoffPolicy = null
+    ) {
+        if ($retryPolicy === null) {
+            $retryPolicy = new SimpleRetryPolicy(
+                $maxTries ?? self::DEFAULT_MAX_TRIES,
```

Comment:
Tohle povoluje 0, což je asi správně.

Reply:
Nejsem si jistej, jak to vlastně je. Jestli to udělá jednou a tedy "max tries" je vlastně spíš "max retries". Udělá to a pak 0 až XX krát to zkusí retrynout. Nebo dostane ukol a zkusí ho 1 až XX krát udělat. 

Tj. když je "maxTries" == 0, spustí se to vůbec někdy?